File: OOP-and-Functional-Programming_Samples/OOP/Simulated_Assembly_instructions_and_Samples/c++/mul.cc
#include "mul.h"
#include "Programa.h"
#include <stack>
using namespace std;

void Mul::ejecutar(Programa* programa){
    stack<int>& pila = programa->getPila();

    int c = pila.top();
    pila.pop();
    int h = pila.top();
    pila.pop();

    pila.push(c * h);
    programa->inc();
}

string Mul::mostrar(){
    return "mul";
}

File: OOP-and-Functional-Programming_Samples/OOP/Simulated_Assembly_instructions_and_Samples/java/Mul.java
public class Mul extends Instruccion{

    public void ejecutar(Programa programa){

        int c = programa.getPila().pop();
        int h = programa.getPila().pop();
        programa.getPila().push(c*h);

        programa.inc();
    }

    public String mostrar(){
        return "mul";
    }
}